front-end/src/app/app.component.ts

```diff
@@ -1,35 +1,126 @@
-import { Component, Input } from '@angular/core';
+import { Component, Input, inject } from '@angular/core';
 import { Task } from 'src/model/Task';
+import { FormControl, FormGroup, ReactiveFormsModule } from '@angular/forms';
+import { CommonModule } from '@angular/common';
+import { TaskCardComponent } from './task-card/task-card.component';
+import { TaskService } from './task.service';
 
 @Component({
   selector: 'app-root',
-  templateUrl: './app.component.html',
+  template: `
+    <div class="container dashboard">
+      <form [formGroup]="applyForm" (submit)="submitTask()">
+        <mat-form-field class="task-form-field">
+          <mat-label>New Task</mat-label>
+          <input matInput type="text" #task_title formControlName="title" />
+        </mat-form-field>
+        <mat-form-field class="task-form-field">
+          <mat-label>Comment</mat-label>
+          <textarea
+            matInput
+            type="text"
+            #task_comment
+            formControlName="comment"
+          ></textarea>
+        </mat-form-field>
+
+        <button
+          mat-raised-button
+          (click)="task_title.value = ''"
+          (click)="task_comment.value = ''"
+          color="primary"
+          class="add_btn"
+          style="display: block; padding: 10px"
+          type="submit"
+        >
+          Add Task
+        </button>
+      </form>
+      <div class="row">
+        <div class="col-sm-5 task-container" *ngIf="gotTaskArray">
+          <h3 style="color: #3f51b5; text-decoration: underline">Tasks:</h3>
+          <app-task-card
+            *ngFor="let task of taskArray"
+            [task]="task"
+            [isDone]="true"
+            (doneEvent)="doneTask($event)"
+            (deleteEvent)="deleteTask($event)"
+          ></app-task-card>
+        </div>
+        <div class="col-1"></div>
+        <div class="col-sm-5 done-container" *ngIf="!doneArrayEmpty">
+          <h3 style="color: #3f51b5; text-decoration: underline">Done:</h3>
+          <app-task-card
+            *ngFor="let task of doneArray"
+            [task]="task"
+            [isDone]="false"
+          ></app-task-card>
+          <button
+            mat-raised-button
+            color="primary"
+            (click)="doneClear()"
+            id="clear"
+          >
+            Clear
+          </button>
+        </div>
+      </div>
+    </div>
+  `,
   styleUrls: ['./app.component.css'],
 })
 export class AppComponent {
+  taskService = inject(TaskService);
   taskArray: Task[] = [];
   doneArray: Task[] = [];
-  renderClearbtn = false;
+  doneArrayEmpty = true;
+  gotTaskArray = false;
```

Comment:
a clean code betartása (ergo a kód olvashatósága) végett, a boolean típusú változókat tipikusan úgy nevezünk el, hogy már a névből kitalálható legyen, hogy 2 állapota van. 

Pl.: a `doneArrayEmpty` majdnem elfogadható, de a nevezéktani konvenciók helyett intuitívebb lenne az `isDoneArrayEmpty`. 

A `gotTaskArray` is majdnem jó, de a jelenlegi kódodban én inkább úgy nevezném el, hogy `shouldShowTaskArray`. Mivel egy bind-olva van egy HTML element-re, ezért a shouldShow tisztábban mutat arra, hogy valami HTML-es mókáért felel ez a változó

itt egy jó [cikk](https://medium.com/swlh/naming-conventions-101-for-developers-8997bb96fd60) ami taglalja ezeket